docs/docs/creating-dynamically-rendered-navigation.md

```diff
@@ -2,7 +2,175 @@
+
+```js
+query SiteQuery {
+  site {
+    siteMetadata {
+      title
+      menuLinks {
+        name
+        link
+      }
+    }
+  }
+}
+``` 
+
+When executing this query within the GraphiQL editor you see output that looks similar to the following:
+
+```js
+{
+  "data": {
+    "site": {
+      "siteMetadata": {
+        "title": "Gatsby Default Starter",
+        "menuLinks": [
+          {
+            "name": "home",
+            "link": "/"
+          },
+          {
+            "name": "page2",
+            "link": "/page-2"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+Perfect! You now have a way of obtaining data from the `gatsby-config.js` file,  let's continue by pulling this data into the layout using the query you just formed.
+
+### Pulling data inside the layout component
+
+Inside your project locate the `src/components` folder and navigate to the `layout.js` file. Within this layout component you should notice a component named `StaticQuery`. 
+
+StaticQuery is a new component introduced in Gatsby V2 and allows you to run GraphQL queries within your components, not just pages. It allows developers to colocate data with their components.
+
+Currently the layout component should look like the following:
+
+```js
+  <StaticQuery
+    query={graphql`
+      query SiteTitleQuery {
+        site {
+          siteMetadata {
+            title
+            menuLinks {
+              name
+              link
+            }
+          }
+        }
+      }
+    `} ...left out for brevity/>
+```
+
+Let's extend the query within this component to include the menuLinks meta data, so it looks as so: (**Note:** *The GraphQL query must be located within the StaticQuery component, rather than referencing a variable from elsewhere due to how Gatsby parses our project*).
+
+```js
+<StaticQuery query={
+      graphql`
+      query SiteQuery {
+        site {
+          siteMetadata {
+            title
+            menuLinks {
+              name
+              link
+            }
+          }
+        }
+      }
+    `} 
+...left out for brevity />
+```
+
+With the above changes to your `StaticQuery` component, the  `render` property which accepts a function that takes one argument now has  has access to the siteMetadata for use inside the function (as the argument). The last thing that is left to do is to display the site's navigation.
+
+To do this the header component that is already available in the project seems like it might be a good starting place to display the navigation. Lets pass the menuLinks to this header component as props like so:
```

Comment:
remember to format `menuLinks` with backticks here and in the rest of the document